pynsot/commands/cmd_protocol_types.py

```diff
@@ -0,0 +1,340 @@
+)
+@click.pass_context
+def add(ctx, required_attributes, description, id, name, site_id):
+    """
+    Add a new Protocol Type.
+
+    You must provide a Protocol Type name or ID using the UPDATE option.
+
+    When adding a new Protocol Type, you must provide a value for the -n/--name
+    option.
+
+    Examples: OSPF, BGP, etc.
+
+    You must also provide attributes, you may specify the -a/--attributes
+    option once for each key/value pair.
+
+    You must provide a Site ID using the -s/--site-id option.
+    """
+    data = ctx.params
+
+    # Remove if empty; allow default assignment
+    if description is None:
+        data.pop('description')
+
+    ctx.obj.add(data)
+
+
+# List
+@cli.group(invoke_without_command=True)
+@click.option(
+    '-r',
+    '--required_attributes',
+    metavar='REQUIRED ATTRIBUTES',
+    help='A key/value pair attached to this Protocol Type (format: key=value).',
+    multiple=True,
+    callback=callbacks.transform_attributes,
+)
+@click.option(
+    '-e',
+    '--description',
+    metavar='DESCRIPTION',
+    type=str,
+    help='Filter by Protocol Type matching this description.',
+)
+@click.option(
+    '-g',
+    '--grep',
+    is_flag=True,
+    help='Display list results in a grep-friendly format.',
+    default=False,
+    show_default=True,
+)
+@click.option(
+    '-I',
+    '--id',
+    metavar='ID',
+    help='Unique ID of the Protocol Type being retrieved.',
+)
+@click.option(
+    '-n',
+    '--name',
+    metavar='NAME',
+    help='Filter to Protocol Type matching this name.'
+)
+@click.option(
+    '-s',
+    '--site-id',
+    metavar='SITE_ID',
+    help='Unique ID of the Site this Protocol Type is under.',
+    callback=callbacks.process_site_id,
+)
+@click.pass_context
+def list(ctx, required_attributes, description, grep, id, name, site_id):
+    """
+    List existing Protocol Types for a Site.
+
+    You must provide a Site ID using the -s/--site-id option.
+
+    When listing Protocol Types, all objects are displayed by default. You
+    optionally may lookup a single Protocol Types by ID using the -i/--id option.
+    The ID can either be the numeric ID of the Protocol Type.
+    """
+    data = ctx.params
+
+    # If we provide ID, show more fields
+    if id is not None or name:
+        display_fields = VERBOSE_FIELDS
+    else:
+        display_fields = DISPLAY_FIELDS
+
+    # If we aren't passing a sub-command, just call list(), otherwise let it
+    # fallback to default behavior.
+    if ctx.invoked_subcommand is None:
+        ctx.obj.list(
+            data, display_fields=display_fields,
+            verbose_fields=VERBOSE_FIELDS
+        )
+
+@list.command()
+@click.pass_context
+def protocols(ctx, *args, **kwargs):
+    """Recursively get all protocols of an Protocol Type."""
+    callbacks.list_subcommand(
+        ctx, display_fields=PROTOCOL_DISPLAY_FIELDS, my_name=ctx.info_name
+    )
+
+
+# Remove
+@cli.command()
+@click.option(
+    '-I',
```

Comment:
`-I` -> `-i`

Also, this should be a `type=types.NATURAL_KEY` field since `ProtocolType.name` is the natural key for `ProtocolType`.